src/Features/Core/Portable/SolutionCrawler/IncrementalAnalyzerBase.cs

```diff
@@ -29,6 +29,8 @@ public virtual Task DocumentResetAsync(Document document, CancellationToken canc
         public virtual bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs e)
             => false;
 
+        public virtual bool IsDocumentAnalysisDependentOnItBeingActiveDocumentOrNot => false;
```

Comment:
i had virtual defaults :)   i have no confidence all the right subclasses override this appropriately.

Reply:
Code has been removed.